api/swagger.yml

```diff
@@ -1698,6 +1698,49 @@ components:
         - part_number
         - etag
 
+    CopyPartSource:
+      type: object
+      properties:
+        repository:
+          type: string
+        ref:
+          type: string
+        path:
+          type: string
+        range:
+          description: "Range of bytes to copy"
+          type: string
+          pattern: '^bytes=((\d*-\d*,? ?)+)$'
+      required:
+        - repository
+        - ref
+        - path
+
+    UploadPartFrom:
+      type: object
+      properties:
+        type:
+          description: "Future versions may allow operations other than copy"
+          type: string
+          pattern: '^(?:copy|)$'
```

Comment:
it is required and use regex to match possible value which we can specify using enum.
Would making it an optional enum make more sense?